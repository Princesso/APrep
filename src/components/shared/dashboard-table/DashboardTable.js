import React from 'react'
import ViewParcel from '../modal/ViewParcel'
import NewParcel from '../modal/NewParcel'
import ChangeDestination from '../modal/ChangeDestination'
import {getUserParcels} from '../../../actions/userActions'
import {Link, withRouter} from 'react-router-dom'
import {connect} from 'react-redux'
import {toast} from 'react-toastify'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTruck, faCheck, faClock, faBan } from "@fortawesome/free-solid-svg-icons";
import './dashboardtable.css'

export class DashboardTable extends React.Component {
	constructor() {
    super()
    this.state = { viewParcel: false, createNew: false, selectedParcel: {}}
    this.toggleCreateNew = this.toggleCreateNew.bind(this)
		this.toggleViewParcel = this.toggleViewParcel.bind(this)
		
	}

  toggleCreateNew() {
    this.setState({createNew: !this.state.createNew})
	}

	setSelectedParcel(id) {
		this.setState({selectedParcel: this.props.userParcels.find(item => item.id === id )})
	}

	toggleViewParcel() {
    this.setState({viewParcel: !this.state.viewParcel})
	} 
	
	componentDidMount() { 
		this.props.getUserParcels()
		.then(()=>{
				toast.success('You have not created any parcles yet')
		})
		.catch((error)=> {
			toast.error('An error occurred when trying to get your parcels')
		})
	}

  render() {
		const parcels = this.props.userParcels;
		const total = this.props.userParcels.length;
		const canceledParcels = this.props.userParcels.filter(parcel => parcel.status == 'canceled').length
		const pendingParcels = this.props.userParcels.filter(parcel => parcel.status == 'pending').length
		const deliveredParcels = this.props.userParcels.filter(parcel => parcel.status == 'delivered').length
		const listOfParcels = parcels.map((userParcel, index) => {
			return(
					<tr key={index}>
						<td>{userParcel.id}</td>
						<td>{userParcel.senton.slice(0,10)}</td>
						<td>{userParcel.itemname}</td>
						<td>{userParcel.currentlocation}</td>
						<td>{userParcel.toaddress}</td>
						<td>{userParcel.recipient}</td>
						<td>{userParcel.status}</td>
						<td>
							<button
								onClick={() => {
									this.toggleViewParcel(),
									this.setSelectedParcel(userParcel.id)
								}}
							>
								View
							</button></td>
					</tr>
			)
		})

		return(
			<div>
				<NewParcel toggleModalView={this.toggleCreateNew} visible={this.state.createNew} />
				<ViewParcel toggleModalView={this.toggleViewParcel} visible={this.state.viewParcel} selectedUserParcel={this.state.selectedParcel}/>
				<div className="dashcontainer">
					<div className="welcome">
						<h5>Welcome <span id="username"></span></h5>
						
						<button id="new-delivery" onClick={() => this.toggleCreateNew()}>New Delivery Order</button>
					</div>
					<div className="cards">
						<div className="card">
							<div id="delivery-truck">
									<span className="fas fa-truck"><FontAwesomeIcon icon={faTruck} /></span>
							</div>
							<div>
									<h6>Orders</h6>
									<p id="total-orders">{total }</p>
							</div>
						</div>
						<div className="card">
								<div id="delivery-truck">
										<span className="fas fa-check"><FontAwesomeIcon icon={faCheck} /></span>
								</div>
								<div>
										<h6>Delivered</h6>
										<p id="total-delivered">{deliveredParcels}</p>
								</div>
						</div>
						<div className="card">
								<div id="delivery-truck">
										<span className="fas fa-clock"><FontAwesomeIcon icon={faClock} /></span>
								</div>
								<div>
										<h6>Pending</h6>
										<p id="total-pending">{pendingParcels}</p>
								</div>
						</div>
						<div className="card">
								<div id="delivery-truck">
										<span className="fas fa-ban"><FontAwesomeIcon icon={faBan} /></span>
								</div>
								<div>
										<h6>Canceled</h6>
										<p id="total-canceled">{canceledParcels}</p>
								</div>
						</div>
					</div>
					<div id="all">
							All Orders
							<p id="dashboard-error"></p>
					</div>
					<div className="table-container">
						<table className="table" id="dashboard-table">
							<thead>
								<tr>
										<th data-name="id">ID</th>
										<th data-name="id">SENT-ON</th>
										<th data-name="itemname">ITEM NAME</th>
										<th data-name="currentlocation">CURRENT LOCATION</th>
										<th data-name="toaddress">DESTINATION</th>
										<th data-name="recipient">RECIPIENT</th>
										<th data-name="status">STATUS</th>
										<th data-name="action">ACTION</th>
									</tr>
							</thead>
							<tbody>
								{ listOfParcels }
							</tbody>	
						</table>
					</div>
				</div>
			</div>
		)
	}
  
}

const mapStateToProps = state => ({
  userParcels: state.userParcels || []
})

export default connect(mapStateToProps, {getUserParcels}) (withRouter(DashboardTable))