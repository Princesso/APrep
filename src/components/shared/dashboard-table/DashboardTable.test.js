import React from 'react';
import DashboardTable from './DashboardTable';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import Store from '../../../store';
import { BrowserRouter, Route, Switch } from 'react-router-dom'

test('Dashboard table', () => {
  const component = renderer.create(
    <Provider store={Store}>
      <BrowserRouter>
          <Switch>
            <DashboardTable/>
          </Switch>
      </BrowserRouter>
    </Provider>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})