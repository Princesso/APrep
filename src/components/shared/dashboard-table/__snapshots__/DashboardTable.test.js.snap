// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard table 1`] = `
<div>
  <div />
  <div />
  <div
    className="dashcontainer"
  >
    <div
      className="welcome"
    >
      <h5>
        Welcome 
        <span
          id="username"
        />
      </h5>
      <button
        id="new-delivery"
        onClick={[Function]}
      >
        New Delivery Order
      </button>
    </div>
    <div
      className="cards"
    >
      <div
        className="card"
      >
        <div
          id="delivery-truck"
        >
          <span
            className="fas fa-truck"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-truck fa-w-20 "
              data-icon="truck"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M624 352h-16V243.9c0-12.7-5.1-24.9-14.1-33.9L494 110.1c-9-9-21.2-14.1-33.9-14.1H416V48c0-26.5-21.5-48-48-48H48C21.5 0 0 21.5 0 48v320c0 26.5 21.5 48 48 48h16c0 53 43 96 96 96s96-43 96-96h128c0 53 43 96 96 96s96-43 96-96h48c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zM160 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm320 0c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-208H416V144h44.1l99.9 99.9V256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </div>
        <div>
          <h6>
            Orders
          </h6>
          <p
            id="total-orders"
          >
            2
          </p>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          id="delivery-truck"
        >
          <span
            className="fas fa-check"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check fa-w-16 "
              data-icon="check"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </div>
        <div>
          <h6>
            Delivered
          </h6>
          <p
            id="total-delivered"
          >
            0
          </p>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          id="delivery-truck"
        >
          <span
            className="fas fa-clock"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-clock fa-w-16 "
              data-icon="clock"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </div>
        <div>
          <h6>
            Pending
          </h6>
          <p
            id="total-pending"
          >
            1
          </p>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          id="delivery-truck"
        >
          <span
            className="fas fa-ban"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-ban fa-w-16 "
              data-icon="ban"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </div>
        <div>
          <h6>
            Canceled
          </h6>
          <p
            id="total-canceled"
          >
            1
          </p>
        </div>
      </div>
    </div>
    <div
      id="all"
    >
      All Orders
      <p
        id="dashboard-error"
      />
    </div>
    <div
      className="table-container"
    >
      <table
        className="table"
        id="dashboard-table"
      >
        <thead>
          <tr>
            <th
              data-name="id"
            >
              ID
            </th>
            <th
              data-name="id"
            >
              SENT-ON
            </th>
            <th
              data-name="itemname"
            >
              ITEM NAME
            </th>
            <th
              data-name="currentlocation"
            >
              CURRENT LOCATION
            </th>
            <th
              data-name="toaddress"
            >
              DESTINATION
            </th>
            <th
              data-name="recipient"
            >
              RECIPIENT
            </th>
            <th
              data-name="status"
            >
              STATUS
            </th>
            <th
              data-name="action"
            >
              ACTION
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              2019-03-26
            </td>
            <td>
              bag
            </td>
            <td>
              Abuja
            </td>
            <td>
              Lagos
            </td>
            <td />
            <td>
              pending
            </td>
            <td>
              <button
                onClick={[Function]}
              >
                View
              </button>
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              2019-03-26
            </td>
            <td>
              bag
            </td>
            <td>
              Abuja
            </td>
            <td>
              Lagos
            </td>
            <td />
            <td>
              canceled
            </td>
            <td>
              <button
                onClick={[Function]}
              >
                View
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
